### PR TITLE
bert encoder inherits from nn.Module now

### PR DESCRIPTION
**Pull request by raceee**
I was using MLX to do some masked language modeling using bert model example, and I wasn't able to

```optimizer.update(model.trainable_parameters(), grads)```

Like how it is done in the MLP example documentation [here](https://ml-explore.github.io/mlx/build/html/examples/mlp.html)

After adding ```super().__init__()``` to both the ```BERT``` class and the ```BertEmbeddings``` my training loop proceeded.

### CODE REVIEW

**Review by awni — APPROVED** [review]
Oops! Good catch!
